mitodl/mit-open
# Micromasters ETL

# What are the relevant tickets?
Closes #143 

# Description (What does it do?)
- Moves the micromasters ETL from the `course_catalog` app to `learning_resources` and refactors it to use the new models.
- Moves associated tasks and mgmt command
- Cleans up a few other ETL mgmt commands

# How can this be tested?
- Populate the following .env settings, use heroku for values.
  ```
  EDX_API_URL
  EDX_API_CLIENT_ID
  EDX_API_CLIENT_SECRET
  EDX_API_ACCESS_TOKEN_URL
  EDX_LEARNING_COURSE_BUCKET_NAME
  MICROMASTERS_CATALOG_API_URL=https://micromasters.mit.edu/api/v0/catalog/
  MICROMASTERS_COURSE_URL=https://micromasters.mit.edu/api/v0/courseruns/ 
  ```
- Run `./manage.py backpopulate_mit_edx_data`
- Run `./manage.py backpopulate_micromasters_data`
- Check that you have some populated Micromasters programs with associated courses, most of those courses/runs should have been imported via backpopulate_mit_edx_data but there might be some stray new ones created from the micromasters pipeline.
